thrift.js

```diff
@@ -58,6 +59,45 @@ var messageExceptionTypesDef = require('./message').exceptionTypesDef;
 var validThriftIdentifierRE = /^[a-zA-Z_][a-zA-Z0-9_\.]+$/;
 
 function Thrift(options) {
+    this._init(options);
+
+    if (this.asyncLoad) {
+        return;
+    }
+
+    this._parse(this.filename, this.allowIncludeAlias);
+    this._compileAndLink();
+}
+
+// Alternative constructor allowing for asynchronous source loading.
+Thrift.load = function load(options, cb) {
+    if (!options) {
+        return cb(Error('options required'));
+    } else if (typeof options !== 'object') {
+        return cb(Error('options must be object'));
+    } else if (!options.fs || !options.fs.load) {
```

Comment:
It occurs to me that fs.load should be fs.readFile since this would make it possible to pass the Node.js fs module without alteration to do async loading server side.

Reply:
Agreed!